File: files/2024-06-23/CSEC 22.06.py
import pandas as pd

path = 'files/Excel Challenge  22nd June.xlsx'
input = pd.read_excel(path, skiprows = 1, usecols= "B")
test  = pd.read_excel(path, skiprows = 1, usecols= "D", nrows = 3)

input = pd.concat([input.iloc[::2].reset_index(drop=True), input.iloc[1::2].reset_index(drop=True)], axis=1)
input.columns = ['Dates', 'Values']
filtered_data = input[input['Values'] < input['Values'].shift()]
result = filtered_data[['Dates']].astype("datetime64[ns]").reset_index(drop=True)

print(result.equals(test)) # True

File: files/2024-06-23/CSEC 22.06.R
library(tidyverse)
library(readxl)

path = "files/Excel Challenge  22nd June.xlsx"

input = read_xlsx(path, range = "B2:B22", col_types = "text")
test  = read_xlsx(path, range = "D2:D5")

result <- tibble(
  Dates = as.POSIXct(as.Date(as.numeric(input$`Date & Orders`[seq(1, nrow(input), 2)]), 
                             origin = "1899-12-30")),
  Values = as.numeric(input$`Date & Orders`[seq(2, nrow(input), 2)])
) %>%
  filter(Values < lag(Values)) %>%
  select(Dates)

identical(result, test)
# [1] TRUE



